### PR TITLE
Update index page

### PR DESCRIPTION
**Pull request by alex-ju**
This PR updates the index page:
 - uses the table component to render document list
 - uses the document-list component to render document link and details in the first cell for each row
 - updates markup for styling and layout 
 - ~~allows sorting by state~~

[Trello card](https://trello.com/c/caR0G5Tu)

### Preview
screen shot 2018-09-24 at 12 00 52

### CODE REVIEW

File: app/views/documents/index/_results.html.erb [hunk 1]
```diff
@@ -1,49 +1,57 @@
...
+    state_active = @sort.delete_prefix("-") == "publication_state"
+    state_link_to_asc = state_active && @sort[0] == "-"
+
+    state_sort_direction = "descending" if state_active && state_link_to_asc
+    state_sort_direction = "ascending" if state_active && !state_link_to_asc
+  %>
+
+  <%= t.head do %>
+    <%= t.header t("documents.index.search_results.headings.title") %>
+    <%= t.header t("documents.index.search_results.headings.state"), {
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/views/documents/index/_results.html.erb]: I don't think it brings values, added it as a column as per design and to test it works with columns having multiple filters. I'll be moving it back in the first column, but still not sure which will be the other columns though.

File: app/views/documents/index/_results.html.erb [hunk 2]
```diff
@@ -1,49 +1,57 @@
...
-          %>
-        </td>
-
-        <td class="govuk-table__cell ">
-          <%= document.updated_at.strftime("%d %b %Y") %>
-        </td>
-      </tr>
+
+      <% document_item = capture do %>
+      <%= render "govuk_publishing_components/components/document_list", {
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/views/documents/index/_results.html.erb]: Right now it brings visual consistency and is flexible enough to add any attributes for a document, that's why I didn't want to build a new one. I'm thinking that if we feel the need to style it differently we can do with overwrites, if we need different markup we can move to a new component.

File: app/views/documents/index/_results.html.erb [hunk 3]
```diff
@@ -1,49 +1,57 @@
...
-          class: classes
-        ) %>
-      </th>
-    </tr>
-  </thead>
-
-  <tbody class="govuk-table__body">
+<%= GovukPublishingComponents::AppHelpers::TableHelper.helper(self, "Documents list", { sortable: true, caption_classes: "govuk-visually-hidden" }) do |t| %>
+
+  <%
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/views/documents/index/_results.html.erb]: Updated the code for the other two comments. Not sure how to refactor the filter though.

File: app/views/documents/index.html.erb [hunk 4]
```diff
@@ -1,32 +1,30 @@
...
-    <br/>
+    <hr class="govuk-section-break govuk-section-break--visible govuk-section-break--l">
 
     <div class="govuk-body">
-      <strong> <%= t("documents.index.guidance.title") %></strong>
-      <br/>
-      <%= link_to "Planning, writing and managing content",
-        "https://www.gov.uk/topic/government-digital-guidance/content-publishing", class: "govuk-link" %>
+      <p class="govuk-heading-s"> <%= t("documents.index.guidance.title") %></p>
+      <p><%= link_to "Planning, writing and managing content", "https://www.gov.uk/topic/government-digital-guidance/content-publishing", class: "govuk-link" %></p>
```
Comment (benthorner) [inline app/views/documents/index.html.erb]: Think we're missing 'target: _blank' here ;-).

Comment (kevindew) [reply app/views/documents/index.html.erb]: Enjoy your holiday Ben!

File: config/locales/en/documents/index.yml [hunk 5]
```diff
@@ -7,6 +7,7 @@ en:
       search_results:
         headings:
           title: Document title
+          state: State
           last_updated: Last updated
           last_updated_asc: Sort by last updated ascending
```
Comment (benthorner) [inline config/locales/en/documents/index.yml]: I think these two aren't used anymore, so can probably go.

**Pull request merged**